cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -683,38 +693,43 @@ runGenesisCreateStaked (GenesisDir rootdir)
   forM_ (zip [ 1 .. bulkPoolCredFiles ] bulkSlices) $
     uncurry (writeBulkPoolCredentials pooldir)
 
-  forM_ [ 1 .. genNumStDelegs ] $ \index ->
-    createDelegatorCredentials stdeldir index
-
-  delegations :: [Delegation] <-
-    -- Distribute M delegates across N pools:
-    forM [ (pool, delegIx)
-         | (pool, poolIx) <- zip pools [1 ..]
-         , delegIxLocal <- [ 1 .. delegsPerPool ] ++
-                           -- Add all remaining delegates to the last pool:
-                           if delegsRemaining /= 0 && poolIx == genNumPools
-                           then [ delegsPerPool + 1 .. delegsPerPool + delegsRemaining ]
-                           else []
-         , let delegIx = delegIxLocal + delegsPerPool * (poolIx - 1)] $
-      uncurry (computeDelegation network stdeldir)
+  let (delegsPerPool, delegsRemaining) = divMod genNumStDelegs genNumPools
+      delegsForPool poolIx = if delegsRemaining /= 0 && poolIx == genNumPools
+        then delegsPerPool
+        else delegsPerPool + delegsRemaining
+      distribution = [pool | (pool, poolIx) <- zip pools [1 ..], _ <- [1 .. delegsForPool poolIx]]
+
+  -- Distribute M delegates across N pools:
+  delegations <- liftIO $ Lazy.forM distribution $ computeDelegation network
```

Comment:
I would say lets just implement out own version of `forM` and `replicateM` in the cli and remove the `hw-lazy` dependency

Reply:
`cardano-node` doesn't seem the right place to put them given how generic they are.  Can  I put them into `cardano-base`?

Reply:
`cardano-prelude` might be a good place for those functions and tests, unless the overall plan is to stop depending on `cardano-prelude` (at least that is what we are striking for in `cardano-base` and `cardano-ledger` repos)